### PR TITLE
Fix asPercent error when series has more values than totalSeries

### PR DESCRIPTION
**Pull request by carrieedwards**
This PR fixes an error that has been found when the length of the series values is greater than the length of the totalSeries values that are being used to calculate the percentage. The error is thrown because the index that is attempting to be accessed in the totalSeries is out of range.

### CODE REVIEW

File: expr/functions/asPercent/function.go [hunk 1]
```diff
@@ -61,6 +61,31 @@ func sumSeries(seriesList []*types.MetricData) *types.MetricData {
...
+	if math.IsNaN(seriesValue) || math.IsNaN(totalValue) || totalValue == 0 {
+		value = math.NaN()
+	} else {
+		value = seriesValue * (100 / totalValue)
+	}
+	return value
+}
+
+func getPercentages(series, totalSeries *types.MetricData) {
+	if len(series.Values) > len(totalSeries.Values) {
```
Comment (leizor) [inline expr/functions/asPercent/function.go:75]: Is there any chance that `len(series.Values) < len(totalSeries.Values)`?

Comment (carrieedwards) [reply expr/functions/asPercent/function.go:75]: Yes, I think that could be possible

Comment (carrieedwards) [reply expr/functions/asPercent/function.go:75]: That's a good point. I will add support for that case.

Comment (leizor) [reply expr/functions/asPercent/function.go:75]: I'm wondering if we should be growing `series.Values` to the same length as `totalSeries.Values` in that case; I think that might be [what graphite-web is doing](https://github.com/grafana/graphite-web/blob/master/webapp/graphite/render/functions.py#L1046)?

Comment (carrieedwards) [reply expr/functions/asPercent/function.go:75]: Yes, NaN should be added to series until they are the same length. Adding that now

**Review by leizor — APPROVED** [review]

File: expr/functions/asPercent/function.go [hunk 2]
```diff
@@ -61,6 +61,39 @@ func sumSeries(seriesList []*types.MetricData) *types.MetricData {
...
+	var value float64
+	if math.IsNaN(seriesValue) || math.IsNaN(totalValue) || totalValue == 0 {
+		value = math.NaN()
+	} else {
+		value = seriesValue * (100 / totalValue)
+	}
+	return value
+}
+
+func getPercentages(series, totalSeries *types.MetricData) {
```
Comment (ywwg) [inline expr/functions/asPercent/function.go:74]: this is tricky enough that we should include a docstring.

**Review by npazosmendez — APPROVED** [review]

**Pull request merged**